### PR TITLE
Support python 3.9 for TensorFlow and MLFlow

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Optuna will support the latest TensorFlow and Keras by https://github.com/optuna/optuna/pull/2852.

## Description of the changes

- Add configuration for the tests with python 3.9 for tensorflow and mlflow
- Remove a part of version constraints introduced by https://github.com/optuna/optuna-examples/pull/44
- Use tensorflow.keras instead of keras in mlflow since the existing keras example does not work anymore with keras alone

For backwards compatibility, I keep the version constraint of `/keras` codes.

### CODE REVIEW

File: keras/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,4 @@
-# TODO(hvy): Remove the version constraint. See https://github.com/optuna/optuna/pull/2851
-keras<2.6.0
+keras
```
Comment (nzw0301) [inline keras/requirements.txt:1]: note: we can remove this line since keras is installed via the installation of Tensorflow.

**Comment by nzw0301** [comment]
I think we have room to revise the following warning messages:
- https://github.com/optuna/optuna-examples/blob/6a6b20ad634627eebb3e7e104f73b70b45c6e624/keras/keras_integration.py#L104-L111
- https://github.com/optuna/optuna-examples/blob/6a6b20ad634627eebb3e7e104f73b70b45c6e624/keras/keras_simple.py#L84-L91

File: keras/keras_integration.py [hunk 2]
```diff
@@ -60,7 +62,7 @@ def create_model(trial):
     lr = trial.suggest_float("lr", 1e-5, 1e-1, log=True)
     model.compile(
         loss="categorical_crossentropy",
-        optimizer=keras.optimizers.RMSprop(lr=lr),
+        optimizer=RMSprop(lr=lr),
```
Comment (nzw0301) [inline keras/keras_integration.py:65]: Note that `lr` is deprecated; `learning_rate` is supposed to be appropriate.

File: keras/keras_simple.py [hunk 3]
```diff
@@ -87,7 +87,9 @@ def objective(trial):
         "There is now only one Keras: tf.keras. "
         "There may be some breaking changes for some workflows by upgrading to keras 2.4.0. "
         "Test before upgrading. "
-        "REF:https://github.com/keras-team/keras/releases/tag/2.4.0"
+        "REF: https://github.com/keras-team/keras/releases/tag/2.4.0. "
+        "There is an alternative callback function that can be used instead: "
+        ":class:`~optuna.integration.TFKerasPruningCallback`",
```
Comment (hvy) [inline keras/keras_simple.py:92]: This is not an pruning/callback example but should we still keep this addition?

**Review by hvy — COMMENTED** [review]
Thanks a lot for this follow up, looks great besides a small nitpick.

File: keras/keras_simple.py (hunk 3 above)
Comment (nzw0301) [reply keras/keras_simple.py:92]: Thank you for your feedback! I agree with you. I'll revert this change.

File: keras/keras_integration.py (hunk 2 above)
Comment (hvy) [reply keras/keras_integration.py]: Do you want to follow that up in a separate PR? I'd happily review if you're okay with that.

**Review by hvy — APPROVED** [review]
Thank you as always, LGTM!

File: keras/keras_integration.py (hunk 2 above)
Comment (nzw0301) [reply keras/keras_integration.py]: Thank you for kind pinging me. Sure, I'll send a follow-up PR.

**Pull request merged**